game/core/view_strategies/rpg_movement.py

```diff
@@ -201,22 +210,60 @@ def search(self):
 
             for sprite in sprites_in_range:
 
-                if "item" in sprite.properties:
-                    self.player_sprite.add_item_to_inventory(sprite)
+                if "name" in sprite.properties:
+                    key = self.player_sprite.add_item_to_inventory(sprite)
+                    self.gui.draw_message_box(
+                        message=f"{sprite.properties['name']} added to inventory!",
+                        notes=f"Press {key} to use",
+                    )
                     self.state.remove_sprite_from_map(sprite, True)
 
-    def animate_player_item(self, player_sprite):
-        config = constants.ITEM_CONFIG[player_sprite.item.properties["item"]][
+    def use_item(self, slot):
+        inventory = self.player_sprite.inventory
+        if len(inventory) < slot:
+            self.gui.draw_message_box(message=f"No item in inventory slot {slot}")
+            return
+        index = slot - 1
+        item_name = inventory[index].properties["name"]
+        # Build raft
+        if item_name in RAFT_COMPONENTS:
+            has_missing_components = check_missing_components(inventory)
+            if has_missing_components:
+                missing_components_text = generate_missing_components_text(inventory)
+                self.gui.draw_message_box(
+                    message=missing_components_text["message"],
+                    notes=missing_components_text["notes"],
+                    seconds=5,
+                )
+            else:
+                # TODO spawn raft
+                self.gui.draw_message_box(
+                    message="Raft is not implemented yet, check back later!"
+                )
```

Comment:
We could have the message box display the content of a file managed by the feedback tool so that when `pytest` passes for the task, the message box disappears, but while the task is failing, the message would appear?

Maybe after MS2, we could have the unit tests runnable from the UI, but probably best to have them run separately and see what users think

Reply:
I think this could work better as part of the separate "task messages" in the gui that's pinned on the top right until it's resolved. What do you think?